api/v1alpha1/stage_types.go

```diff
@@ -621,6 +621,11 @@ type AnalysisTemplateReference struct {
 	//
 	// +kubebuilder:validation:Required
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// ClusterScope determines whether the template is an
+	// AnalysisTemplate or a ClusterAnalysisTemplate resource
+	//
+	// +kubebuilder:validation:Optional
+	ClusterScope bool `json:"clusterScope" protobuf:"varint,2,opt,name=clusterScope"`
```

Comment:
I would prefer to have this be a `kind` selector, which matches the way we reference `ClusterPromotionTask` resources at present:

https://github.com/akuity/kargo/blob/ef223531d8c4f052aa5ca3aff264684fc97ebd4a/api/v1alpha1/promotion_types.go#L124-L140

Reply:
I'll get on that, I just took the method used by Argo Rollouts when referencing the different template kinds but I see you way make more sense in this project.

Reply:
@hiddeco I have changed the clusterscope to use a kind reference instead.